test-runner-poc/lib/filters/mongodb_version_filter.js

```diff
@@ -14,66 +15,30 @@ const f = require('util').format;
  * }
  */
 class MongoDBVersionFilter {
+  initializeFilter(callback) {
+    const mongoClient = new MongoClient(process.env.MONGODB_URI || 'mongodb://127.0.0.1:27017');
+    mongoClient.connect((err, client) => {
+      client.db('admin').command({buildInfo: true}, (err, result) => {
+        this.version = result.version;
+        client.close(callback);
+      });
+    })
+
+  }
+
   constructor(options) {
     this.options = options || {};
-    this.version = null;
   }
 
-  beforeStart(configuration, callback) {
-    const self = this;
-    if (this.options.skip) {
-      return callback();
-    }
 
-    if (configuration.type === 'core') {
-      configuration.newConnection({ w: 1 }, function(err, topology) {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        topology.command(f('%s.$cmd', configuration.db), { buildInfo: true }, function(
-          commandErr,
-          result
-        ) {
-          if (commandErr) throw commandErr;
-          self.version = result.result.version;
-          console.log('running against mongodb version:');
-          console.dir(result.result);
-
-          topology.destroy();
-          callback();
-        });
-      });
-    } else {
-      configuration.newClient({ w: 1 }).connect(function(err, client) {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        client.db('admin').command({ buildInfo: true }, function(_err, result) {
-          if (_err) {
-            callback(_err);
-            return;
-          }
-
-          self.version = result.versionArray.slice(0, 3).join('.');
-          console.log('running against mongodb version:');
-          console.dir(result);
-
-          client.close();
-          callback();
-        });
-      });
-    }
-  }
 
   filter(test) {
     if (this.options.skip) return true;
+    if (!test) return true;
```

Comment:
a cool thing you can do here is take advantage of lazy evaluation in the `&&` operator:

```js
if (! (test && test.metadata && test.metdata.requires && test.metadata.requires.mongodb) ) {
  return true;
}
```